clang/lib/CConv/RewriteUtils.cpp

```diff
@@ -816,16 +816,49 @@ static void emit(Rewriter &R, ASTContext &C, std::set<FileID> &Files,
         }
 }
 
+typedef enum {
+  IS_UNCHECKED,
+  IS_CHECKED,
+} AnnotationNeeded;
+
 
-// This Visitor adds _Checked and _UnChecked annotations to blocks
 class CheckedRegionAdder : public clang::RecursiveASTVisitor<CheckedRegionAdder>
 {
   public:
     explicit CheckedRegionAdder(ASTContext *_C, Rewriter &_R, ProgramInfo &_I,
-                                std::set<llvm::FoldingSetNodeID> &S)
-      : Context(_C), Writer(_R), Info(_I), Seen(S) {
+                                std::map<llvm::FoldingSetNodeID, AnnotationNeeded> &M)
+      : Context(_C), Writer(_R), Info(_I), Map(M) {}
 
+    bool VisitCompoundStmt(CompoundStmt *S) {
```

Comment:
What is this boolean communicating, on return? This is a visitor method, so presumably it's telling the visitor something about the traversal?